index.js

```diff
@@ -2441,6 +2441,24 @@
         return n < 0 || negativeZero(n) ? Nothing() : slice(0, -n, xs);
       });
 
+  //# reverse :: [a] -> [a]
+  //.
+  //. Takes an array and returns an array of the its elements in
+  //. reversed order.
+  //.
+  //. ```javascript
+  //. > S.reverse([1, 2, 3])
+  //. [3, 2, 1]
+  //. ```
+  S.reverse =
+  def('reverse',
+      {},
+      [List(a), List(a)],
+      function(xs) {
+          return R.type(xs) === 'String' ? xs.split('').reverse().join('')
+                                 : xs.slice().reverse();
```

Comment:
Since `{'0': 'foo', '1': 'bar', length: 2} :: List String`, it's not safe to assume the existence of a `slice` method. Something along these lines would be more appropriate:

```javascript
function reverse(xs) {
  if (_type(xs) === 'String') return reverse(xs.split('')).join('');
  var result = [];
  for (var idx = xs.length - 1; idx >= 0; idx -= 1) result.push(xs[idx]);
  return result;
}
```